### PR TITLE
AURORA: Add loader for Scaleform GFx files

### PR DESCRIPTION
**Pull request by Nostritius**
As the name says, this PR introduces a loader for scaleform gfx files, used by dragonage and dragonage2. I still does not implement all of the necessary tags, but the most relevant ones.

### CODE REVIEW

File: src/aurora/gfxfile.h [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+public:
+	/**
+	 * Open a gfx file specified by the reference and used with the specified avm.
+	 * @param resref the resref of the gfx file
+	 * @param avm The AVM that should be used throughout the loading process
+	 */
+	GFXFile(const Common::UString &resref, Aurora::ActionScript::AVM &avm);
+
+	/** Get the framerate for this gfx file. */
+	float getFrameRate(){ return _frameRate; };
```
Comment (DrMcCoy) [inline src/aurora/gfxfile.h]: Add a space between `()` and `{`, please

File: src/aurora/gfxfile.cpp [hunk 2]
```diff
@@ -0,0 +1,1137 @@
...
+}
+
+void GFXCharacter::getShape(Shape &shape) const {
+	if (_type != kShape)
+		throw Common::Exception("Character is not a shape");
+
+	shape = boost::get<Shape>(_value);
+}
+
+void
```
Comment (DrMcCoy) [inline src/aurora/gfxfile.cpp]: Remove the linebreak here, please

File: src/aurora/gfxfile.cpp [hunk 3]
```diff
@@ -0,0 +1,1137 @@
...
+GFXCharacter::getFont(Font &font) const {
+	if (_type != kFont)
+		throw Common::Exception("Character is not a font");
+
+	font = boost::get<Font>(_value);
+}
+
+void
```
Comment (DrMcCoy) [inline src/aurora/gfxfile.cpp]: Remove the linebreak here, please

File: src/aurora/gfxfile.cpp [hunk 4]
```diff
@@ -0,0 +1,1137 @@
...
+	_characters.insert(std::make_pair(shapeId, GFXCharacter::createShape(shapeId, shape)));
+}
+
+void GFXFile::readBackgroundColor() {
+	readRGB();
+}
+
+void GFXFile::readExportAssets() {
+	uint16 count = _gfx->readUint16LE();
+	for (int i = 0; i < count; ++i) {
```
Comment (DrMcCoy) [inline src/aurora/gfxfile.cpp]: signed/unsigned

File: src/aurora/gfxfile.cpp [hunk 5]
```diff
@@ -0,0 +1,1137 @@
...
+	const Common::UString url = Common::FilePath::getStem(readNullTerminatedString());
+	GFXFile import(url, avm);
+
+	byte reserved = _gfx->readByte();
+	assert(reserved == 1);
+	reserved = _gfx->readByte();
+	assert(reserved == 0);
+
+	uint16 count = _gfx->readUint16LE();
+	for (int i = 0; i < count; ++i) {
```
Comment (DrMcCoy) [inline src/aurora/gfxfile.cpp]: signed/unsigned

File: src/aurora/gfxfile.cpp [hunk 6]
```diff
@@ -0,0 +1,1137 @@
...
+		warning("GFXFile::readDefineFont() Invalid language code");
+
+	font.name = readLengthPrefixedString();
+
+	uint16 numGlyphs = _gfx->readUint16LE();
+
+	size_t startOffsetTable = _gfx->pos();
+
+	std::vector<uint32> offsetTable;
+	for (int i = 0; i < numGlyphs; ++i) {
```
Comment (DrMcCoy) [inline src/aurora/gfxfile.cpp]: signed/unsigned, here and multiple times below

File: src/aurora/gfxfile.cpp [hunk 7]
```diff
@@ -0,0 +1,1137 @@
...
+	assert(bitmapFormat == 14);
+
+	byte prefixLen = _gfx->readByte();
+	assert(prefixLen == 0);
+
+	Common::UString swfName = readLengthPrefixedString();
+
+	if (_gfx->pos() - oldPos < header.tagLength) {
+		uint16 numCodeOffsets = _gfx->readUint16LE();
+		for (int i = 0; i < numCodeOffsets; ++i) {
```
Comment (DrMcCoy) [inline src/aurora/gfxfile.cpp]: signed/unsigned

**Comment by Nostritius** [comment]
The issues should now be fixed. I also fixed some things on my own, mostly making some variables const

**Comment by DrMcCoy** [comment]
Merged, thanks! :)

**Pull request merged**